### PR TITLE
React edit

### PR DESCRIPTION
**Pull request by jisie**
In the `REACT_FORMS.md` chapter:
1. Advanced challenge to implement edit has been removed
1. Dropdown for employees has been updated to use `employeeId`. Same change is reflected in code snippet for Kandy Korner.

New chapter, `REACT_EDIT.md`, has been added. README for book 4 has this chapter listed as chapter 11.

### CODE REVIEW

File: book-4-the-apprentice/chapters/REACT_EDIT.md [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+  }}
+/>
+```
+
+## Update method in ApplicationViews
+
+When an animal's information is edited, the HTTP request will be a PUT. Define a method in your `AnimalManager` for the fetch call. This method will take two arguments: the id of the animal that will be updated as well as an object that contains the new information for the animal that we want to save to the database.
+
+```jsx
+put(animalId, editedAnimal) {
```
Comment (stevebrownlee) [inline book-4-the-apprentice/chapters/REACT_EDIT.md]: Can you change the method signature to just accept the `animal` object? The `id` property can be snagged from the object (see more below).

File: book-4-the-apprentice/chapters/REACT_EDIT.md [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+      },
+      body: JSON.stringify(editedAnimal)
+    }).then(data => data.json());
+  }
+```
+
+In your `ApplicationViews`, you are going to add a method that uses our newly defined `put` method in our `AnimalManger`. Just like with our CREATE and DELETE, any time the data is changed, another fetch to get the new data is made and we update state.
+
+```jsx
+updateAnimal = (animalId, editedAnimalObject) => {
```
Comment (stevebrownlee) [inline book-4-the-apprentice/chapters/REACT_EDIT.md]: Just `editedAnimalObject` as a single argument.

File: book-4-the-apprentice/chapters/REACT_EDIT.md [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+        this.setState(stateToChange)
+    }
+
+    updateExistingAnimal = evt => {
+      evt.preventDefault()
+
+      if (this.state.employee === "") {
+        window.alert("Please select a caretaker");
+      } else {
+        const editedAnimal = {
```
Comment (stevebrownlee) [inline book-4-the-apprentice/chapters/REACT_EDIT.md]: Put the primary key in the object.

```
const editedAnimal = {
          id: this.props.match.params.animalId,
          name: this.state.animalName,
          breed: this.state.breed,
          employeeId: parseInt(this.state.employeeId)
        };
```

File: book-4-the-apprentice/chapters/REACT_EDIT.md [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+          name: this.state.animalName,
+          breed: this.state.breed,
+          employeeId: parseInt(this.state.employeeId)
+        };
+
+    this.props.updateAnimal(this.props.match.params.animalId, editedAnimal)
```
Comment (stevebrownlee) [inline book-4-the-apprentice/chapters/REACT_EDIT.md]: Just send `editedAnimal` as single argument

**Comment by jisie** [comment]
Refactored so both the `put` and `updateAnimal` have a single argument.

**Comment by stevebrownlee** [comment]
:shipit:

**Pull request merged**